
From: Arnd Bergmann <arnd@arndb.de>

I found that the prototypes for sys_waitid and sys_fcntl in
<linux/syscalls.h> don't match the implementation.  In order to keep all
prototypes in sync in the future, now include the header from each file
implementing any syscall.

Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/pci/syscall.c  |    1 +
 25-akpm/fs/aio.c               |    1 +
 25-akpm/fs/buffer.c            |    1 +
 25-akpm/fs/dcache.c            |    1 +
 25-akpm/fs/dcookies.c          |    1 +
 25-akpm/fs/fcntl.c             |    3 ++-
 25-akpm/fs/filesystems.c       |    1 +
 25-akpm/fs/ioctl.c             |    1 +
 25-akpm/fs/locks.c             |    1 +
 25-akpm/fs/namei.c             |    1 +
 25-akpm/fs/namespace.c         |    1 +
 25-akpm/fs/nfsctl.c            |    1 +
 25-akpm/fs/open.c              |    1 +
 25-akpm/fs/quota.c             |    1 +
 25-akpm/fs/read_write.c        |    1 +
 25-akpm/fs/readdir.c           |    1 +
 25-akpm/fs/select.c            |    1 +
 25-akpm/fs/stat.c              |    1 +
 25-akpm/fs/super.c             |    1 +
 25-akpm/fs/xattr.c             |    1 +
 25-akpm/init/do_mounts_devfs.c |    1 -
 25-akpm/ipc/mqueue.c           |    1 +
 25-akpm/ipc/msg.c              |    1 +
 25-akpm/ipc/sem.c              |    1 +
 25-akpm/ipc/shm.c              |    1 +
 25-akpm/kernel/acct.c          |    1 +
 25-akpm/kernel/capability.c    |    1 +
 25-akpm/kernel/exec_domain.c   |    1 +
 25-akpm/kernel/exit.c          |    1 +
 25-akpm/kernel/itimer.c        |    1 +
 25-akpm/kernel/panic.c         |    1 -
 25-akpm/kernel/posix-timers.c  |    1 +
 25-akpm/kernel/printk.c        |    1 +
 25-akpm/kernel/sched.c         |    1 +
 25-akpm/kernel/signal.c        |    1 +
 25-akpm/kernel/sys.c           |    1 +
 25-akpm/kernel/sysctl.c        |    1 +
 25-akpm/kernel/time.c          |    1 +
 25-akpm/kernel/timer.c         |    1 +
 25-akpm/mm/fadvise.c           |    1 +
 25-akpm/mm/filemap.c           |    1 +
 25-akpm/mm/fremap.c            |    1 +
 25-akpm/mm/madvise.c           |    1 +
 25-akpm/mm/mincore.c           |    1 +
 25-akpm/mm/mlock.c             |    1 +
 25-akpm/mm/mprotect.c          |    1 +
 25-akpm/mm/mremap.c            |    1 +
 25-akpm/mm/msync.c             |    1 +
 25-akpm/mm/nommu.c             |    1 +
 25-akpm/mm/swapfile.c          |    1 +
 include/linux/syscalls.h       |    0 
 51 files changed, 49 insertions(+), 3 deletions(-)

diff -puN drivers/pci/syscall.c~add-missing-linux-syscallsh-includes drivers/pci/syscall.c
--- 25/drivers/pci/syscall.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/drivers/pci/syscall.c	Mon Sep 20 14:58:36 2004
@@ -11,6 +11,7 @@
 #include <linux/errno.h>
 #include <linux/pci.h>
 #include <linux/smp_lock.h>
+#include <linux/syscalls.h>
 #include <asm/uaccess.h>
 
 
diff -puN fs/aio.c~add-missing-linux-syscallsh-includes fs/aio.c
--- 25/fs/aio.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/aio.c	Mon Sep 20 14:58:36 2004
@@ -14,6 +14,7 @@
 #include <linux/time.h>
 #include <linux/aio_abi.h>
 #include <linux/module.h>
+#include <linux/syscalls.h>
 
 #define DEBUG 0
 
diff -puN fs/buffer.c~add-missing-linux-syscallsh-includes fs/buffer.c
--- 25/fs/buffer.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/buffer.c	Mon Sep 20 14:58:36 2004
@@ -20,6 +20,7 @@
 
 #include <linux/config.h>
 #include <linux/kernel.h>
+#include <linux/syscalls.h>
 #include <linux/fs.h>
 #include <linux/mm.h>
 #include <linux/percpu.h>
diff -puN fs/dcache.c~add-missing-linux-syscallsh-includes fs/dcache.c
--- 25/fs/dcache.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/dcache.c	Mon Sep 20 14:58:36 2004
@@ -15,6 +15,7 @@
  */
 
 #include <linux/config.h>
+#include <linux/syscalls.h>
 #include <linux/string.h>
 #include <linux/mm.h>
 #include <linux/fs.h>
diff -puN fs/dcookies.c~add-missing-linux-syscallsh-includes fs/dcookies.c
--- 25/fs/dcookies.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/dcookies.c	Mon Sep 20 14:58:36 2004
@@ -13,6 +13,7 @@
  */
 
 #include <linux/config.h>
+#include <linux/syscalls.h>
 #include <linux/module.h>
 #include <linux/slab.h>
 #include <linux/list.h>
diff -puN fs/fcntl.c~add-missing-linux-syscallsh-includes fs/fcntl.c
--- 25/fs/fcntl.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/fcntl.c	Mon Sep 20 14:58:36 2004
@@ -4,6 +4,7 @@
  *  Copyright (C) 1991, 1992  Linus Torvalds
  */
 
+#include <linux/syscalls.h>
 #include <linux/init.h>
 #include <linux/mm.h>
 #include <linux/fs.h>
@@ -362,7 +363,7 @@ static long do_fcntl(int fd, unsigned in
 	return err;
 }
 
-asmlinkage long sys_fcntl(int fd, unsigned int cmd, unsigned long arg)
+asmlinkage long sys_fcntl(unsigned int fd, unsigned int cmd, unsigned long arg)
 {	
 	struct file *filp;
 	long err = -EBADF;
diff -puN fs/filesystems.c~add-missing-linux-syscallsh-includes fs/filesystems.c
--- 25/fs/filesystems.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/filesystems.c	Mon Sep 20 14:58:36 2004
@@ -6,6 +6,7 @@
  *  table of configured filesystems
  */
 
+#include <linux/syscalls.h>
 #include <linux/fs.h>
 #include <linux/slab.h>
 #include <linux/kmod.h>
diff -puN fs/ioctl.c~add-missing-linux-syscallsh-includes fs/ioctl.c
--- 25/fs/ioctl.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/ioctl.c	Mon Sep 20 14:58:36 2004
@@ -5,6 +5,7 @@
  */
 
 #include <linux/config.h>
+#include <linux/syscalls.h>
 #include <linux/mm.h>
 #include <linux/smp_lock.h>
 #include <linux/file.h>
diff -puN fs/locks.c~add-missing-linux-syscallsh-includes fs/locks.c
--- 25/fs/locks.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/locks.c	Mon Sep 20 14:58:36 2004
@@ -122,6 +122,7 @@
 #include <linux/security.h>
 #include <linux/slab.h>
 #include <linux/smp_lock.h>
+#include <linux/syscalls.h>
 #include <linux/time.h>
 
 #include <asm/semaphore.h>
diff -puN fs/namei.c~add-missing-linux-syscallsh-includes fs/namei.c
--- 25/fs/namei.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/namei.c	Mon Sep 20 14:58:36 2004
@@ -25,6 +25,7 @@
 #include <linux/smp_lock.h>
 #include <linux/personality.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 #include <linux/mount.h>
 #include <linux/audit.h>
 #include <asm/namei.h>
diff -puN fs/namespace.c~add-missing-linux-syscallsh-includes fs/namespace.c
--- 25/fs/namespace.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/namespace.c	Mon Sep 20 14:58:36 2004
@@ -9,6 +9,7 @@
  */
 
 #include <linux/config.h>
+#include <linux/syscalls.h>
 #include <linux/slab.h>
 #include <linux/sched.h>
 #include <linux/smp_lock.h>
diff -puN fs/nfsctl.c~add-missing-linux-syscallsh-includes fs/nfsctl.c
--- 25/fs/nfsctl.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/nfsctl.c	Mon Sep 20 14:58:36 2004
@@ -13,6 +13,7 @@
 #include <linux/linkage.h>
 #include <linux/namei.h>
 #include <linux/mount.h>
+#include <linux/syscalls.h>
 #include <asm/uaccess.h>
 
 /*
diff -puN fs/open.c~add-missing-linux-syscallsh-includes fs/open.c
--- 25/fs/open.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/open.c	Mon Sep 20 14:58:36 2004
@@ -22,6 +22,7 @@
 #include <asm/uaccess.h>
 #include <linux/fs.h>
 #include <linux/pagemap.h>
+#include <linux/syscalls.h>
 
 #include <asm/unistd.h>
 
diff -puN fs/quota.c~add-missing-linux-syscallsh-includes fs/quota.c
--- 25/fs/quota.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/quota.c	Mon Sep 20 14:58:36 2004
@@ -13,6 +13,7 @@
 #include <linux/kernel.h>
 #include <linux/smp_lock.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 
 /* Check validity of quotactl */
 static int check_quotactl_valid(struct super_block *sb, int type, int cmd, qid_t id)
diff -puN fs/readdir.c~add-missing-linux-syscallsh-includes fs/readdir.c
--- 25/fs/readdir.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/readdir.c	Mon Sep 20 14:58:36 2004
@@ -14,6 +14,7 @@
 #include <linux/fs.h>
 #include <linux/dirent.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 #include <linux/unistd.h>
 
 #include <asm/uaccess.h>
diff -puN fs/read_write.c~add-missing-linux-syscallsh-includes fs/read_write.c
--- 25/fs/read_write.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/read_write.c	Mon Sep 20 14:58:36 2004
@@ -13,6 +13,7 @@
 #include <linux/dnotify.h>
 #include <linux/security.h>
 #include <linux/module.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/unistd.h>
diff -puN fs/select.c~add-missing-linux-syscallsh-includes fs/select.c
--- 25/fs/select.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/select.c	Mon Sep 20 14:58:36 2004
@@ -14,6 +14,7 @@
  *     of fds to overcome nfds < 16390 descriptors limit (Tigran Aivazian).
  */
 
+#include <linux/syscalls.h>
 #include <linux/module.h>
 #include <linux/slab.h>
 #include <linux/smp_lock.h>
diff -puN fs/stat.c~add-missing-linux-syscallsh-includes fs/stat.c
--- 25/fs/stat.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/stat.c	Mon Sep 20 14:58:36 2004
@@ -14,6 +14,7 @@
 #include <linux/fs.h>
 #include <linux/namei.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/unistd.h>
diff -puN fs/super.c~add-missing-linux-syscallsh-includes fs/super.c
--- 25/fs/super.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/super.c	Mon Sep 20 14:58:36 2004
@@ -32,6 +32,7 @@
 #include <linux/buffer_head.h>		/* for fsync_super() */
 #include <linux/mount.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 #include <linux/vfs.h>
 #include <linux/writeback.h>		/* for the emergency remount stuff */
 #include <linux/idr.h>
diff -puN fs/xattr.c~add-missing-linux-syscallsh-includes fs/xattr.c
--- 25/fs/xattr.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/fs/xattr.c	Mon Sep 20 14:58:36 2004
@@ -13,6 +13,7 @@
 #include <linux/xattr.h>
 #include <linux/namei.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 #include <asm/uaccess.h>
 
 /*
diff -puN include/linux/syscalls.h~add-missing-linux-syscallsh-includes include/linux/syscalls.h
diff -puN init/do_mounts_devfs.c~add-missing-linux-syscallsh-includes init/do_mounts_devfs.c
--- 25/init/do_mounts_devfs.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/init/do_mounts_devfs.c	Mon Sep 20 14:58:36 2004
@@ -2,7 +2,6 @@
 #include <linux/kernel.h>
 #include <linux/dirent.h>
 #include <linux/string.h>
-#include <linux/syscalls.h>
 
 #include "do_mounts.h"
 
diff -puN ipc/mqueue.c~add-missing-linux-syscallsh-includes ipc/mqueue.c
--- 25/ipc/mqueue.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/ipc/mqueue.c	Mon Sep 20 14:58:36 2004
@@ -22,6 +22,7 @@
 #include <linux/msg.h>
 #include <linux/skbuff.h>
 #include <linux/netlink.h>
+#include <linux/syscalls.h>
 #include <net/sock.h>
 #include "util.h"
 
diff -puN ipc/msg.c~add-missing-linux-syscallsh-includes ipc/msg.c
--- 25/ipc/msg.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/ipc/msg.c	Mon Sep 20 14:58:36 2004
@@ -24,6 +24,7 @@
 #include <linux/list.h>
 #include <linux/security.h>
 #include <linux/sched.h>
+#include <linux/syscalls.h>
 #include <asm/current.h>
 #include <asm/uaccess.h>
 #include "util.h"
diff -puN ipc/sem.c~add-missing-linux-syscallsh-includes ipc/sem.c
--- 25/ipc/sem.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/ipc/sem.c	Mon Sep 20 14:58:36 2004
@@ -71,6 +71,7 @@
 #include <linux/time.h>
 #include <linux/smp_lock.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 #include <asm/uaccess.h>
 #include "util.h"
 
diff -puN ipc/shm.c~add-missing-linux-syscallsh-includes ipc/shm.c
--- 25/ipc/shm.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/ipc/shm.c	Mon Sep 20 14:58:36 2004
@@ -26,6 +26,7 @@
 #include <linux/proc_fs.h>
 #include <linux/shmem_fs.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 #include <asm/uaccess.h>
 
 #include "util.h"
diff -puN kernel/acct.c~add-missing-linux-syscallsh-includes kernel/acct.c
--- 25/kernel/acct.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/acct.c	Mon Sep 20 14:58:36 2004
@@ -53,6 +53,7 @@
 #include <linux/vfs.h>
 #include <linux/jiffies.h>
 #include <linux/times.h>
+#include <linux/syscalls.h>
 #include <asm/uaccess.h>
 #include <asm/div64.h>
 #include <linux/blkdev.h> /* sector_div */
diff -puN kernel/capability.c~add-missing-linux-syscallsh-includes kernel/capability.c
--- 25/kernel/capability.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/capability.c	Mon Sep 20 14:58:36 2004
@@ -10,6 +10,7 @@
 #include <linux/mm.h>
 #include <linux/module.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 #include <asm/uaccess.h>
 
 unsigned securebits = SECUREBITS_DEFAULT; /* systemwide security settings */
diff -puN kernel/exec_domain.c~add-missing-linux-syscallsh-includes kernel/exec_domain.c
--- 25/kernel/exec_domain.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/exec_domain.c	Mon Sep 20 14:58:36 2004
@@ -14,6 +14,7 @@
 #include <linux/module.h>
 #include <linux/personality.h>
 #include <linux/sched.h>
+#include <linux/syscalls.h>
 #include <linux/sysctl.h>
 #include <linux/types.h>
 
diff -puN kernel/exit.c~add-missing-linux-syscallsh-includes kernel/exit.c
--- 25/kernel/exit.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/exit.c	Mon Sep 20 14:58:36 2004
@@ -24,6 +24,7 @@
 #include <linux/mount.h>
 #include <linux/proc_fs.h>
 #include <linux/mempolicy.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/unistd.h>
diff -puN kernel/itimer.c~add-missing-linux-syscallsh-includes kernel/itimer.c
--- 25/kernel/itimer.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/itimer.c	Mon Sep 20 14:58:36 2004
@@ -9,6 +9,7 @@
 #include <linux/mm.h>
 #include <linux/smp_lock.h>
 #include <linux/interrupt.h>
+#include <linux/syscalls.h>
 #include <linux/time.h>
 
 #include <asm/uaccess.h>
diff -puN kernel/panic.c~add-missing-linux-syscallsh-includes kernel/panic.c
--- 25/kernel/panic.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/panic.c	Mon Sep 20 14:58:36 2004
@@ -16,7 +16,6 @@
 #include <linux/notifier.h>
 #include <linux/init.h>
 #include <linux/sysrq.h>
-#include <linux/syscalls.h>
 #include <linux/interrupt.h>
 #include <linux/nmi.h>
 
diff -puN kernel/posix-timers.c~add-missing-linux-syscallsh-includes kernel/posix-timers.c
--- 25/kernel/posix-timers.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/posix-timers.c	Mon Sep 20 14:58:36 2004
@@ -43,6 +43,7 @@
 #include <linux/compiler.h>
 #include <linux/idr.h>
 #include <linux/posix-timers.h>
+#include <linux/syscalls.h>
 #include <linux/wait.h>
 #include <linux/workqueue.h>
 
diff -puN kernel/printk.c~add-missing-linux-syscallsh-includes kernel/printk.c
--- 25/kernel/printk.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/printk.c	Mon Sep 20 14:58:36 2004
@@ -30,6 +30,7 @@
 #include <linux/smp.h>
 #include <linux/security.h>
 #include <linux/bootmem.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 
diff -puN kernel/sched.c~add-missing-linux-syscallsh-includes kernel/sched.c
--- 25/kernel/sched.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/sched.c	Mon Sep 20 14:58:36 2004
@@ -42,6 +42,7 @@
 #include <linux/percpu.h>
 #include <linux/kthread.h>
 #include <linux/seq_file.h>
+#include <linux/syscalls.h>
 #include <linux/times.h>
 #include <asm/tlb.h>
 
diff -puN kernel/signal.c~add-missing-linux-syscallsh-includes kernel/signal.c
--- 25/kernel/signal.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/signal.c	Mon Sep 20 14:58:36 2004
@@ -20,6 +20,7 @@
 #include <linux/tty.h>
 #include <linux/binfmts.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 #include <linux/ptrace.h>
 #include <asm/param.h>
 #include <asm/uaccess.h>
diff -puN kernel/sys.c~add-missing-linux-syscallsh-includes kernel/sys.c
--- 25/kernel/sys.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/sys.c	Mon Sep 20 14:58:36 2004
@@ -23,6 +23,7 @@
 #include <linux/security.h>
 #include <linux/dcookies.h>
 #include <linux/suspend.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/io.h>
diff -puN kernel/sysctl.c~add-missing-linux-syscallsh-includes kernel/sysctl.c
--- 25/kernel/sysctl.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/sysctl.c	Mon Sep 20 14:58:36 2004
@@ -40,6 +40,7 @@
 #include <linux/times.h>
 #include <linux/limits.h>
 #include <linux/dcache.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/processor.h>
diff -puN kernel/time.c~add-missing-linux-syscallsh-includes kernel/time.c
--- 25/kernel/time.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/time.c	Mon Sep 20 14:58:36 2004
@@ -31,6 +31,7 @@
 #include <linux/timex.h>
 #include <linux/errno.h>
 #include <linux/smp_lock.h>
+#include <linux/syscalls.h>
 #include <asm/uaccess.h>
 #include <asm/unistd.h>
 
diff -puN kernel/timer.c~add-missing-linux-syscallsh-includes kernel/timer.c
--- 25/kernel/timer.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/kernel/timer.c	Mon Sep 20 14:58:36 2004
@@ -31,6 +31,7 @@
 #include <linux/time.h>
 #include <linux/jiffies.h>
 #include <linux/cpu.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/unistd.h>
diff -puN mm/fadvise.c~add-missing-linux-syscallsh-includes mm/fadvise.c
--- 25/mm/fadvise.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/fadvise.c	Mon Sep 20 14:58:36 2004
@@ -15,6 +15,7 @@
 #include <linux/backing-dev.h>
 #include <linux/pagevec.h>
 #include <linux/fadvise.h>
+#include <linux/syscalls.h>
 
 #include <asm/unistd.h>
 
diff -puN mm/filemap.c~add-missing-linux-syscallsh-includes mm/filemap.c
--- 25/mm/filemap.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/filemap.c	Mon Sep 20 14:58:36 2004
@@ -27,6 +27,7 @@
 #include <linux/pagevec.h>
 #include <linux/blkdev.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 /*
  * This is needed for the following functions:
  *  - try_to_release_page
diff -puN mm/fremap.c~add-missing-linux-syscallsh-includes mm/fremap.c
--- 25/mm/fremap.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/fremap.c	Mon Sep 20 14:58:36 2004
@@ -14,6 +14,7 @@
 #include <linux/swapops.h>
 #include <linux/rmap.h>
 #include <linux/module.h>
+#include <linux/syscalls.h>
 
 #include <asm/mmu_context.h>
 #include <asm/cacheflush.h>
diff -puN mm/madvise.c~add-missing-linux-syscallsh-includes mm/madvise.c
--- 25/mm/madvise.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/madvise.c	Mon Sep 20 14:58:36 2004
@@ -7,6 +7,7 @@
 
 #include <linux/mman.h>
 #include <linux/pagemap.h>
+#include <linux/syscalls.h>
 
 
 /*
diff -puN mm/mincore.c~add-missing-linux-syscallsh-includes mm/mincore.c
--- 25/mm/mincore.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/mincore.c	Mon Sep 20 14:58:36 2004
@@ -11,6 +11,7 @@
 #include <linux/pagemap.h>
 #include <linux/mm.h>
 #include <linux/mman.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/pgtable.h>
diff -puN mm/mlock.c~add-missing-linux-syscallsh-includes mm/mlock.c
--- 25/mm/mlock.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/mlock.c	Mon Sep 20 14:58:36 2004
@@ -7,6 +7,7 @@
 
 #include <linux/mman.h>
 #include <linux/mm.h>
+#include <linux/syscalls.h>
 
 
 static int mlock_fixup(struct vm_area_struct * vma, 
diff -puN mm/mprotect.c~add-missing-linux-syscallsh-includes mm/mprotect.c
--- 25/mm/mprotect.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/mprotect.c	Mon Sep 20 14:58:36 2004
@@ -18,6 +18,7 @@
 #include <linux/security.h>
 #include <linux/mempolicy.h>
 #include <linux/personality.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/pgtable.h>
diff -puN mm/mremap.c~add-missing-linux-syscallsh-includes mm/mremap.c
--- 25/mm/mremap.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/mremap.c	Mon Sep 20 14:58:36 2004
@@ -16,6 +16,7 @@
 #include <linux/fs.h>
 #include <linux/highmem.h>
 #include <linux/security.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/cacheflush.h>
diff -puN mm/msync.c~add-missing-linux-syscallsh-includes mm/msync.c
--- 25/mm/msync.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/msync.c	Mon Sep 20 14:58:36 2004
@@ -12,6 +12,7 @@
 #include <linux/mm.h>
 #include <linux/mman.h>
 #include <linux/hugetlb.h>
+#include <linux/syscalls.h>
 
 #include <asm/pgtable.h>
 #include <asm/tlbflush.h>
diff -puN mm/nommu.c~add-missing-linux-syscallsh-includes mm/nommu.c
--- 25/mm/nommu.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/nommu.c	Mon Sep 20 14:58:36 2004
@@ -19,6 +19,7 @@
 #include <linux/vmalloc.h>
 #include <linux/blkdev.h>
 #include <linux/backing-dev.h>
+#include <linux/syscalls.h>
 
 #include <asm/uaccess.h>
 #include <asm/tlb.h>
diff -puN mm/swapfile.c~add-missing-linux-syscallsh-includes mm/swapfile.c
--- 25/mm/swapfile.c~add-missing-linux-syscallsh-includes	Mon Sep 20 14:58:36 2004
+++ 25-akpm/mm/swapfile.c	Mon Sep 20 14:58:36 2004
@@ -25,6 +25,7 @@
 #include <linux/rmap.h>
 #include <linux/security.h>
 #include <linux/backing-dev.h>
+#include <linux/syscalls.h>
 
 #include <asm/pgtable.h>
 #include <asm/tlbflush.h>
_
